### PR TITLE
Bump rubocop to 0.71

### PR DESCRIPTION
**Pull request by abhaynikam**
Rubocop channel 0.71 -> https://github.com/codeclimate/codeclimate-rubocop/pull/183

Shoud we update the gemfile as well with `gem "rubocop", "~> 0.71.0", require: false`?

### CODE REVIEW

**Comment by yahonda** [comment]
I think you will need to update Gemfile and .rubocop.yml file. I'm preparing a similar change at another repository https://github.com/rsim/oracle-enhanced/pull/1887/files 

I rather prefer @koic or someone who is familiar with RuboCop makes this kind of change.

**Comment by abhaynikam** [comment]
@yahonda Alright. @koic or someone who is familiar can pick this up. I'll just try one push and close the PR since I spent sometime on it 😄 . Hope that is fine.

**Comment by yahonda** [comment]
Great. I hope your next one should be fine.

**Comment by abhaynikam** [comment]
@yahonda : Everything passed here. Ping me if I should close this PR.

File: actionview/test/actionpack/abstract/layouts_test.rb [hunk 1]
```diff
@@ -295,7 +295,7 @@ class TestBase < ActiveSupport::TestCase
         10.times do |x|
           controller = WithString.new
           controller.define_singleton_method :index do
-            render template: ActionView::Template::Text.new("Hello string!"), locals: { :"x#{x}" => :omg }
+            render template: ActionView::Template::Text.new("Hello string!"), locals: { "x#{x}" => :omg }
```
Comment (kamipo) [inline actionview/test/actionpack/abstract/layouts_test.rb]: Which cop does this violate?

File: activerecord/test/cases/associations/eager_test.rb [hunk 2]
```diff
@@ -786,19 +786,17 @@ def test_eager_with_has_many_and_limit_and_conditions_on_the_eagers
   def test_eager_with_has_many_and_limit_and_scoped_conditions_on_the_eagers
     posts = nil
     Post.includes(:comments)
-      .where("comments.body like 'Normal%' OR comments.#{QUOTED_TYPE}= 'SpecialComment'")
```
Comment (kamipo) [inline activerecord/test/cases/associations/eager_test.rb]: ditto.

File: actionpack/test/dispatch/request_test.rb [hunk 3]
```diff
@@ -680,8 +680,7 @@ class RequestProtocol < BaseRequestTest
 
 class RequestMethod < BaseRequestTest
   test "method returns environment's request method when it has not been
-    overridden by middleware".squish do
```
Comment (kamipo) [inline actionpack/test/dispatch/request_test.rb]: ditto.

File: actionview/test/actionpack/abstract/layouts_test.rb (hunk 1 above)
Comment (abhaynikam) [reply actionview/test/actionpack/abstract/layouts_test.rb]: `Layout/EmptyLinesAroundBlockBody` -> https://github.com/rails/rails/blob/master/.rubocop.yml#L65

This was changed in https://github.com/rubocop-hq/rubocop/pull/7090 I think

Comment (kamipo) [reply actionview/test/actionpack/abstract/layouts_test.rb]: I asked why rubocop correct intentional symbol key to string key wrongly.

File: .rubocop.yml [hunk 4]
```diff
@@ -77,7 +79,7 @@ Layout/EmptyLinesAroundMethodBody:
 Layout/EmptyLinesAroundModuleBody:
   Enabled: true
 
-Layout/FirstParameterIndentation:
+Layout/IndentFirstArgument:
```
Comment (koic) [inline .rubocop.yml]: I think it is preferable to move in ASCII order.

File: Gemfile [hunk 5]
```diff
@@ -28,8 +28,9 @@ gem "uglifier", ">= 1.3.0", require: false
 # Explicitly avoid 1.x that doesn't support Ruby 2.4+
 gem "json", ">= 2.0.0"
 
-gem "rubocop", ">= 0.47", require: false
+gem "rubocop", "~> 0.71.0", require: false
```
Comment (koic) [inline Gemfile]: rails/rails repository is maintained by Gemfile.lock. Is this change meaningful?

File: actionview/test/actionpack/abstract/layouts_test.rb (hunk 1 above)
Comment (abhaynikam) [reply actionview/test/actionpack/abstract/layouts_test.rb]: @kamipo : My bad. I wanted to comment it for another comment of your(https://github.com/rails/rails/pull/36426#discussion_r291054490).

`Style/HashSyntax` cop fails here. It should be following as per -> https://github.com/rubocop-hq/rubocop/pull/6948

Rubocop new syntax when keys are interpolated.

```
"x#{x}": :omg
```

I'll make that change.

Comment (koic) [reply actionview/test/actionpack/abstract/layouts_test.rb]: This difference is due to the following changes.
https://github.com/rubocop-hq/rubocop/pull/6948

**Comment by koic** [comment]
I think that `Rails: Enabled: true` can be removed from .rubocop.yaml.

```diff
diff --git a/.rubocop.yml b/.rubocop.yml
index 18c20776c6..22161b1e16 100644
--- a/.rubocop.yml
+++ b/.rubocop.yml
@@ -20,9 +20,6 @@ Performance:
   Exclude:
     - '**/test/**/*'

-Rails:
-  Enabled: true
-
 # Prefer assert_not over assert !
 Rails/AssertNot:
   Include:
```

It is enabled when `require: rubocop-rails` is specified in .rubocop.yml.
https://github.com/rubocop-hq/rubocop/pull/7093/files#diff-68891125561f1767ea375a89b05f7e1e

File: activerecord/test/cases/associations/eager_test.rb (hunk 2 above)
Comment (abhaynikam) [reply activerecord/test/cases/associations/eager_test.rb]: https://github.com/rails/rails/pull/36426#discussion_r291056357

Comment (kamipo) [reply activerecord/test/cases/associations/eager_test.rb]: Don't apply your indentation edit.

```
% be rubocop -a activerecord/test/cases/associations/eager_test.rb
Inspecting 1 file
C

Offenses:

activerecord/test/cases/associations/eager_test.rb:792:1: C: [Corrected] Layout/EmptyLinesAroundBlockBody: Extra empty line detected at block body beginning.
activerecord/test/cases/associations/eager_test.rb:801:1: C: [Corrected] Layout/EmptyLinesAroundBlockBody: Extra empty line detected at block body beginning.

1 file inspected, 2 offenses detected, 2 offenses corrected
% git diff activerecord/test/cases/associations/eager_test.rb
diff --git a/activerecord/test/cases/associations/eager_test.rb b/activerecord/test/cases/associations/eager_test.rb
index f7aad9d775..42af1ac8d8 100644
--- a/activerecord/test/cases/associations/eager_test.rb
+++ b/activerecord/test/cases/associations/eager_test.rb
@@ -789,7 +789,6 @@ def test_eager_with_has_many_and_limit_and_scoped_conditions_on_the_eagers
       .where("comments.body like 'Normal%' OR comments.#{QUOTED_TYPE}= 'SpecialComment'")
       .references(:comments)
       .scoping do
-
       posts = authors(:david).posts.limit(2).to_a
       assert_equal 2, posts.size
     end
@@ -798,7 +797,6 @@ def test_eager_with_has_many_and_limit_and_scoped_conditions_on_the_eagers
       .where("authors.name = 'David' AND (comments.body like 'Normal%' OR comments.#{QUOTED_TYPE}= 'SpecialComment')")
       .references(:authors, :comments)
       .scoping do
-
       count = Post.limit(2).count
       assert_equal count, posts.size
     end
```

File: actionpack/test/dispatch/request_test.rb (hunk 3 above)
Comment (kamipo) [reply actionpack/test/dispatch/request_test.rb]:
```
% be rubocop -a actionpack/test/dispatch/request_test.rb
Inspecting 1 file
C

Offenses:

actionpack/test/dispatch/request_test.rb:684:1: C: [Corrected] Layout/EmptyLinesAroundBlockBody: Extra empty line detected at block body beginning.

1 file inspected, 1 offense detected, 1 offense corrected
% git diff actionpack/test/dispatch/request_test.rb
diff --git a/actionpack/test/dispatch/request_test.rb b/actionpack/test/dispatch/request_test.rb
index eb49396145..0ec8dd25e0 100644
--- a/actionpack/test/dispatch/request_test.rb
+++ b/actionpack/test/dispatch/request_test.rb
@@ -681,7 +681,6 @@ class RequestProtocol < BaseRequestTest
 class RequestMethod < BaseRequestTest
   test "method returns environment's request method when it has not been
     overridden by middleware".squish do
-
     ActionDispatch::Request::HTTP_METHODS.each do |method|
       request = stub_request("REQUEST_METHOD" => method)
 
```

**Comment by abhaynikam** [comment]
@kamipo : Done with requested changes.

**Pull request merged**